`nvtx3/nvtx3.hpp` file not found for CPU-only build.

```sh
path/to/kvikio/cpp/include/kvikio/utils.hpp:28:10: fatal error: 'nvtx3/nvtx3.hpp' file not found
   28 | #include <nvtx3/nvtx3.hpp>
      |          ^~~~~~~~~~~~~~~~~
```

As a follow-up, `ifdef`-ing out the include then leads to
```
/path/to/kvikio/cpp/include/kvikio/posix_io.hpp:196:3: error: use of undeclared identifier 'nvtx3'
  196 |   KVIKIO_NVTX_FUNC_RANGE("posix_host_read()", size);
      |   ^
```

--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -145,7 +145,10 @@ if(KvikIO_BUILD_EXAMPLES)
   add_subdirectory(examples)
 endif()
 
-if(KvikIO_BUILD_TESTS AND CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+if(CUDAToolkit_FOUND
+   AND KvikIO_BUILD_TESTS
+   AND CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME
+)
   include(cmake/thirdparty/get_gtest.cmake)
 
   # include CTest module -- automatically calls enable_testing()

--- a/cpp/include/kvikio/utils.hpp
+++ b/cpp/include/kvikio/utils.hpp
@@ -25,7 +25,9 @@
 #include <tuple>
 #include <type_traits>
 
+#ifdef KVIKIO_CUDA_FOUND
 #include <nvtx3/nvtx3.hpp>
+#endif
 
 #include <kvikio/error.hpp>
 #include <kvikio/shim/cuda.hpp>
@@ -300,6 +302,7 @@ struct libkvikio_domain {
 
 // Macro overloads of KVIKIO_NVTX_FUNC_RANGE
 #define KVIKIO_NVTX_FUNC_RANGE_1() NVTX3_FUNC_RANGE_IN(libkvikio_domain)
+#ifdef KVIKIO_CUDA_FOUND
 #define KVIKIO_NVTX_FUNC_RANGE_2(msg, val)                    \
   nvtx3::scoped_range_in<libkvikio_domain> _kvikio_nvtx_range \
   {                                                           \
@@ -308,6 +311,11 @@ struct libkvikio_domain {
       msg, nvtx3::payload { convert_to_64bit(val) }           \
     }                                                         \
   }
+#else
+#define KVIKIO_NVTX_FUNC_RANGE_2(msg, val) \
+  do {                                     \
+  } while (0)
+#endif
 #define GET_KVIKIO_NVTX_FUNC_RANGE_MACRO(_1, _2, NAME, ...) NAME
 
 /**

--- a/cpp/include/kvikio/utils.hpp
+++ b/cpp/include/kvikio/utils.hpp
@@ -293,6 +293,7 @@ inline bool is_future_done(const T& future)
   return future.wait_for(std::chrono::seconds(0)) != std::future_status::timeout;
 }
 
+#ifdef KVIKIO_CUDA_FOUND
 /**
  * @brief Tag type for libkvikio's NVTX domain.
  */
@@ -302,7 +303,6 @@ struct libkvikio_domain {
 
 // Macro overloads of KVIKIO_NVTX_FUNC_RANGE
 #define KVIKIO_NVTX_FUNC_RANGE_1() NVTX3_FUNC_RANGE_IN(libkvikio_domain)
-#ifdef KVIKIO_CUDA_FOUND
 #define KVIKIO_NVTX_FUNC_RANGE_2(msg, val)                    \
   nvtx3::scoped_range_in<libkvikio_domain> _kvikio_nvtx_range \
   {                                                           \
@@ -311,12 +311,8 @@ struct libkvikio_domain {
       msg, nvtx3::payload { convert_to_64bit(val) }           \
     }                                                         \
   }
-#else
-#define KVIKIO_NVTX_FUNC_RANGE_2(msg, val) \
-  do {                                     \
-  } while (0)
-#endif
 #define GET_KVIKIO_NVTX_FUNC_RANGE_MACRO(_1, _2, NAME, ...) NAME
+#endif
 
 /**
  * @brief Convenience macro for generating an NVTX range in the `libkvikio` domain
@@ -337,9 +333,15 @@ struct libkvikio_domain {
  * }
  * ```
  */
+#ifdef KVIKIO_CUDA_FOUND
 #define KVIKIO_NVTX_FUNC_RANGE(...)                                  \
   GET_KVIKIO_NVTX_FUNC_RANGE_MACRO(                                  \
     __VA_ARGS__, KVIKIO_NVTX_FUNC_RANGE_2, KVIKIO_NVTX_FUNC_RANGE_1) \
   (__VA_ARGS__)
+#else
+#define KVIKIO_NVTX_FUNC_RANGE(...) \
+  do {                              \
+  } while (0)
+#endif
 
 }  // namespace kvikio

